runtime/vm/CRIUHelpers.cpp

```diff
@@ -427,6 +435,8 @@ criuRestoreInitializeDump(J9VMThread *currentThread, void *userData, const char
 				*nlsMsgFormat = j9nls_lookup_message(J9NLS_DO_NOT_PRINT_MESSAGE_TAG | J9NLS_DO_NOT_APPEND_NEWLINE,
 						J9NLS_VM_CRIU_RESTORE_INITIALIZE_DUMP_FAILED, NULL);
 				result = FALSE;
+			} else {
+				vm->checkpointState.flags |=  J9VM_CRIU_TRANSITION_TO_DEBUG_INTERPRETER;
```

Comment:
Similar problem here, `criuReloadXDumpAgents()` returning w/o an error doesn't always require `transitionToDebugInterpreter()`.
`criuReloadXDumpAgents()` will need some refactoring to set `J9VM_CRIU_TRANSITION_TO_DEBUG_INTERPRETER` properly.

Reply:
I will explore where in https://github.com/eclipse-openj9/openj9/blob/ecab2f6c958577ff6453a108ab74abf9f00c5691/runtime/rasdump/dmpsup.c#L439 I don't think the solution would be too complex, we just need to determine exactly where to set the flag therein.

For `criuRestoreInitializeTrace` the solution isn't complex at all.

Reply:
Actually, let's go with what Tobi suggested and leave it as it is now. If later we need to discriminate we can then.